### DISCUSSION TITLE
gh-pages does not render anything to the page

### DISCUSSION THREAD

**Discussion by roymansoor** [community/community#47932] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I created vite app and built and deploy it properly. it worked for a while, then when i brought some changes to my page now it doesn't render anything and there is just a white page. i have 2 branches main and gh-pages but still doesn't render anything.
i get "[gh-pages] had recent pushes less than a minute ago "messages. when trying to compare and pull there is nothing to compare. the screenshots are attached.

Appreciate your help know how to solve this problem.

![Screenshot 2023-02-18 at 10 45 54 PM](https://user-images.githubusercontent.com/42701612/219922754-e52808c6-2147-4a35-9862-b80258ad9379.png)

![Screenshot 2023-02-18 at 10 47 05 PM](https://user-images.githubusercontent.com/42701612/219922844-19f5c4d1-f115-40a0-9e77-7a8313864a26.png)

**Comment by melvincwng** [comment, marked as answer]
Hey @roymansoor :wave:, if you are trying to deploy your Vite/React app (i.e. https://roymansoor.github.io/) on GH pages, please take at look at the following below.

Based on Vite's documentation link over [here](https://vitejs.dev/guide/static-deploy.html#github-pages):

![image](https://user-images.githubusercontent.com/77479885/219925351-b95f5a5c-57b3-4a32-929a-559ccb7ad81b.png)

From what I'm seeing in your repository, your current vite.config.js is setup like [this](https://github.com/roymansoor/roymansoor.github.io/blob/main/vite.config.js#L7):

![image](https://user-images.githubusercontent.com/77479885/219925532-bb2dc072-c612-478a-a873-5edb4a2e8977.png)

**It could be an issue with the 'base' key in vite.config.js, hence consider doing the following:**

1. Omitting line 7 - meaning to remove the 'base' key in vite.config.js
2. Run 'npm run build' to generate the new /dist folder/files
3. Commit those files into your code repository
4. Finally redeploy the web app on GitHub pages, and see if it works.
5. Consider looking at the 'References' section below to continue to debug the issue (if the above suggestion doesn't work).


References for your perusal:
1. [Vite Official Documentation](https://vitejs.dev/guide/static-deploy.html#github-pages)
2. [Youtube Video - How to deploy Vite App on GitHub Pages](https://www.youtube.com/watch?v=yo2bMGnIKE8)
3. [Medium Article on Deploying Vite Apps](https://towardsdev.com/deploying-react-application-to-github-pages-with-vite-2d3e32ae97e7)
4. [Dev.to Article on Vite App Deployment](https://dev.to/shashannkbawa/deploying-vite-app-to-github-pages-3ane)
5. [StackOverFlow](https://stackoverflow.com/questions/74518887/blank-page-when-deploying-a-react-app-to-github-pages-and-vite)

Hope this helps and answers your question 😄!

**Reply by roymansoor** [reply]
Hi @melvincwng, 
Thank you for the links and instructions. I have checked my work and confirmed I have done all the steps but still facing the same issue. I made sure the base is properly set. I feel like it has to do with some configuration of GitHub or that specific repo. It was working fine earlier. I will try to delete the repo and create a new one and try again.

**Reply by roymansoor** [reply]
I tried it again with the new repo and all processes, it perfectly works fine when in run preview. but when its deployed its just a blank page. on the console, i get this error (```Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.```)

**Reply by roymansoor** [reply]
@melvincwng, that worked! thanks for the help

**Reply by melvincwng** [reply]
@roymansoor, glad to have helped! 👍🏻

**Comment by ZainilMehta** [comment]
Hi @roymansoor 

I struggled with the same issue for a long! The following solution worked for me and rendered the html pages correctly!
Include index.html as well as the CSS files on the gh-pages orphan branch of the repository!
